python/dolfinx/io/__init__.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2022 Jørgen S. Dokken
+#
+# This file is part of DOLFINx (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+"""Tools for input/output (IO)."""
+
+from dolfinx import cpp as _cpp
+from dolfinx.cpp.io import distribute_entity_data  # noqa: F401
+from dolfinx.io import gmsh  # noqa: F401
```

Comment:
Maybe `from dolfinx.io import gmsh as _gmsh`? I think there is a risk of confusing `gmsh` with the actual Gmsh module.

Reply:
I am simply following the style we use for `dolfinx.fem.petsc`
I think it should be quite clear that this is not the gmsh module, as we in no part of any code import other software as submodules.

I think it would look really weird to use
```python
import dolfinx
dolfinx.io._gmsh.some_function(…)
```

Reply:
The different with PETSc is subtle but is there - in `petsc4py` the name is `PETSc`. We use `petsc`.

I had in mind with the underscore keeping `io.gmsh` private, but if it's imported here only to go into `__all__` that doesn't make sense. I just think it's confusing using the same name as the actual gmsh module.